### PR TITLE
HoC 2024 - Add teacher homepage side-by-side banner

### PR DESCRIPTION
**Pull request by kelbyhawn**
Adds side-by-side banner for Hour of Code 2024 to the teacher homepage at https://studio.code.org/home.

## Links
Jira ticket: [ACQ-2304](https://codedotorg.atlassian.net/browse/ACQ-2304)
Figma mock: [here](https://www.figma.com/design/emvQNdidNnPvoU1NucF7Re/Hour-of-Code-2024?node-id=2710-552&t=MGDc2hQtmlT9lG03-1)

## Testing story
Tested locally

## Followup
We will need to swap out the banner [here](https://github.com/code-dot-org/code-dot-org/blob/6366167326350254d3eff4d43fe3b5dd0ad3339c/dashboard/config/marketing/announcements.json#L3) once the TransformersOne launch is over. I made a ticket: [ACQ-2368](https://codedotorg.atlassian.net/browse/ACQ-2368)

----

Screenshot 2024-09-05 at 11 29 33 AM

### CODE REVIEW

File: dashboard/lib/announcements.rb
Comment (TurnerRiley) [inline dashboard/lib/announcements.rb]: Nit: Would you mind updating the comment here to reflect the updated logic please?

**Review by TurnerRiley — APPROVED** [review]
Love the new test! Just had a quick nit

**Comment by dmcavoy** [comment]
When we put this in it will remove the transformers banner right? Can we wait until closer to 9/20 to merge?

**Comment by kelbyhawn** [comment]
> When we put this in it will remove the transformers banner right? Can we wait until closer to 9/20 to merge?

~@dmcavoy I made a ticket to swap this out on 9/18: [ACQ-2368](https://codedotorg.atlassian.net/browse/ACQ-2368)~ Update: going to merge this on 9/18 or 9/19.

File: dashboard/lib/announcements.rb
Comment (kelbyhawn) [reply dashboard/lib/announcements.rb]: Ah yes, good catch! Updated here: https://github.com/code-dot-org/code-dot-org/pull/60857/commits/b3ac0be0f4aad06ba1d09900d2157b7ffa0121b9

**Comment by kelbyhawn** [comment]
@TurnerRiley can you re-review this please? 🙏  I removed the hoc mode support re: [this Slack update](https://codedotorg.slack.com/archives/C0451MDF70V/p1725636547793939).

**Review by TurnerRiley — APPROVED** [review]
LGTM!

**Review by bethanyaconnor — COMMENTED** [review]
Do we need to merge this so we can get translations?

**Comment by kelbyhawn** [comment]
> Do we need to merge this so we can get translations?

Oh shoot you're right…I always forget about this one! I'll put the TransformersOne banner back here and keep this ticket [ACQ-2368](https://codedotorg.atlassian.net/browse/ACQ-2368).

**Review by TurnerRiley — APPROVED** [review]

**Pull request merged**
